src/main/java/net/dv8tion/jda/api/entities/Template.java

```diff
@@ -0,0 +1,237 @@
+     *         <br>The Template object
+     */
+    @Nonnull
+    static RestAction<Template> resolve(@Nonnull final JDA api, @Nonnull final String code)
+    {
+        return TemplateImpl.resolve(api, code);
+    }
+
+    /**
+     * Syncs this template.
+     * <br>Requires {@link net.dv8tion.jda.api.Permission#MANAGE_SERVER MANAGE_SERVER} in the template's guild.
+     * Will throw an {@link net.dv8tion.jda.api.exceptions.InsufficientPermissionException InsufficientPermissionException} otherwise.
+     *
+     * @throws IllegalStateException
+     *         if the account is not in the template's guild
+     * @throws net.dv8tion.jda.api.exceptions.InsufficientPermissionException
+     *         if the account does not have {@link net.dv8tion.jda.api.Permission#MANAGE_SERVER MANAGE_SERVER} in the template's guild
+     *
+     * @return {@link net.dv8tion.jda.api.requests.RestAction RestAction} - Type: {@link net.dv8tion.jda.api.entities.Template Template}
+     *         <br>The synced Template object
+     */
+    @Nonnull
+    @CheckReturnValue
+    RestAction<Template> sync();
+
+    /**
+     * Deletes this template.
+     * <br>Requires {@link net.dv8tion.jda.api.Permission#MANAGE_SERVER MANAGE_SERVER} in the template's guild.
+     * Will throw an {@link net.dv8tion.jda.api.exceptions.InsufficientPermissionException InsufficientPermissionException} otherwise.
+     *
+     * @throws IllegalStateException
+     *         if the account is not in the template's guild
+     * @throws net.dv8tion.jda.api.exceptions.InsufficientPermissionException
+     *         if the account does not have {@link net.dv8tion.jda.api.Permission#MANAGE_SERVER MANAGE_SERVER} in the template's guild
+     *
+     * @return {@link net.dv8tion.jda.api.requests.RestAction RestAction}
+     */
+    @Nonnull
+    @CheckReturnValue
+    RestAction<Void> delete();
+
+    /**
+     * The template code.
+     *
+     * @return The template code
+     */
+    @Nonnull
+    String getCode();
+
+    /**
+     * The template name.
+     *
+     * @return The template name
+     */
+    @Nonnull
+    String getName();
+
+    /**
+     * The template description.
+     *
+     * @return The template description
+     */
+    @Nullable
+    String getDescription();
+
+    /**
+     * How often this template has been used.
+     *
+     * @return The uses of this template
+     */
+    int getUses();
+
+    /**
+     * The user who created this template.
+     *
+     * @return The user who created this template
+     */
+    @Nonnull
+    User getCreator();
+
+    /**
+     * Returns creation date of this template.
+     *
+     * @return The creation date of this template
+     */
+    @Nonnull
+    OffsetDateTime getTimeCreated();
+
+    /**
+     * Returns the last update date of this template.
+     *
+     * @return The the last update date of this template
+     */
+    @Nonnull
+    OffsetDateTime getTimeUpdated();
```

Comment:
Are you sure this can't be null? I feel like it would be null if its never been updated, no?

Reply:
yes, the field is the date of creation if it's never been synced manually